flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/RecoveredChannelStateHandler.java

```diff
@@ -101,7 +101,10 @@ private RecoveredInputChannel getChannel(InputChannelInfo info) {
 	public void recover(ResultSubpartitionInfo subpartitionInfo, Tuple2<BufferBuilder, BufferConsumer> bufferBuilderAndConsumer) {
 		bufferBuilderAndConsumer.f0.finish();
 		if (bufferBuilderAndConsumer.f1.isDataAvailable()) {
-			getSubpartition(subpartitionInfo).addBufferConsumer(bufferBuilderAndConsumer.f1);
+			boolean added = getSubpartition(subpartitionInfo).add(bufferBuilderAndConsumer.f1);
+			if (!added) {
+				throw new RuntimeException("Buffer consumer couldn't be added to ResultSubpartition");
```

Comment:
`IOException`? (`org.apache.flink.runtime.checkpoint.channel.ChannelStateChunkReader#readChunk` already can handle `IOException`).

Reply:
`false` is returned if the subpartition `isFinished || isReleased`. This case seems to me neither IO error nor recoverable.
I can change to `IOException` though if you insist.

Reply:
I suggested `IOException` because that's what was in the signature of the parent/caller method.

Maybe alternatively `CancelTaskException`, but I'm not entirely sure if that's always the case (what if there was some error in this task, that caused the partition to be released?)

What adds to my confusion is that this return value is ignored in rest of the code base? 🤷‍♂️ 

I think for now it would be the best to go with `IOException`.

Reply:
Changed to `IOException`.

> What adds to my confusion is that this return value is ignored in rest of the code base?

I agree, but I think it can be addressed separately.